src/iologindata.cpp

```diff
@@ -343,6 +343,7 @@ bool IOLoginData::loadPlayer(Player* player, DBResult_ptr result)
 	player->defaultOutfit.lookFeet = result->getNumber<uint16_t>("lookfeet");
 	player->defaultOutfit.lookAddons = result->getNumber<uint16_t>("lookaddons");
 	player->currentOutfit = player->defaultOutfit;
+	player->direction = static_cast<Direction> (result->getNumber<uint16_t>("direction"));
```

Comment:
Direction is **`uint8`** not **`uint16`**.

Reply:
As I said in post earlier I was based on other values just like `lookHead`, `lookBody`, `lookLegs`, `lookFeet` and `lookAddons` which you have 2 lines above. All of these are `uint8_t`, but still from DB is casted as `uint16_t`.

Anyway, I've changed `direction` to `uint8_t`.

Reply:
There may be issues with using uint8_t because it's an unsigned char. It's been a while since I worked in C++, but doing a lexical cast from a char type may turn it into an ASCII character rather than digits. The memory use you're saving on the stack to risk introducing this bug is not even worth having a discussion about, just leave it as uint16_t.

Reply:
Ok, so I reverted my changes. That was my thoughts first that it might not work.